File: SentrySMP.Domain/Entities/PaymentTransaction.cs
using System;

namespace SentrySMP.Domain.Entities
{
    public class PaymentTransaction
    {
        public long Id { get; set; }
        public string Provider { get; set; } = string.Empty; // e.g. PayPal, Stripe
        public string ProviderTransactionId { get; set; } = string.Empty; // order id or session id
        public decimal Amount { get; set; }
        public string Currency { get; set; } = "EUR";
        public string MinecraftUsername { get; set; } = string.Empty;
        public string ItemsJson { get; set; } = string.Empty; // JSON payload of items/cart
        public string Status { get; set; } = string.Empty; // e.g. captured, succeeded
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public string RawResponse { get; set; } = string.Empty; // raw JSON from provider for debugging
    }
}

File: app/Models/PaymentTransaction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class PaymentTransaction extends Model
{
    protected $table = 'payment_transactions';
    public $timestamps = false;

    protected $fillable = [
        'provider', 'provider_transaction_id', 'amount', 'currency',
        'minecraft_username', 'items_json', 'status', 'raw_response', 'created_at',
    ];

    protected $casts = [
        'amount'     => 'decimal:2',
        'created_at' => 'datetime',
    ];

    public function commandQueue()
    {
        return $this->hasMany(CommandQueue::class, 'transaction_id');
    }
}
